Source/AwsCommonRuntimeKit/mqtt/MqttConnection.swift

```diff
@@ -462,7 +442,8 @@ public class MqttConnection {
         }, opCallbackPtr)
 
         defer {
-            pointers.deinitializeAndDeallocate()
+            topicPointer.deinitializeAndDeallocate()
```

Comment:
defer isn't necessary here, topic and payload have already been used

Beginning to feel like we could benefit a ScopedPtr<P> that we can leave on the stack and calls deinitializeAndDeallocate() from its deinit or something

Reply:
interesting idea, so a pointer as a class that deinitializes itself from its own deinit?

Reply:
yes, essentially, it exists on the stack, and when it goes out of scope, it dies. Think of it as an automatically managed temporary